test/e2e/apps/disruption.go

```diff
@@ -217,19 +222,60 @@ var _ = SIGDescribe("DisruptionController", func() {
 		gomega.Expect(err).Should(gomega.MatchError("Cannot evict pod as it would violate the pod's disruption budget."))
 
 		ginkgo.By("Updating the pdb to allow a pod to be evicted")
-		updatePDBMinAvailableOrDie(cs, ns, intstr.FromInt(2))
+		updatePDBMinAvailableOrDie(cs, ns, defaultName, intstr.FromInt(2))
 
 		ginkgo.By("Trying to evict the same pod we tried earlier which should now be evictable")
 		waitForPodsOrDie(cs, ns, 3)
 		err = cs.CoreV1().Pods(ns).Evict(e)
 		framework.ExpectNoError(err) // the eviction is now allowed
+
+		ginkgo.By("Patching the pdb to disallow a pod to be evicted")
+		maxUnavailable := intstr.FromInt(0)
+		patchPDBMinAvailableOrDie(cs, ns, defaultName, nil, &maxUnavailable)
+		pod, err = locateRunningPod(cs, ns) // locate a new running pod
+		framework.ExpectNoError(err)
+		waitForPodsOrDie(cs, ns, 3)
+		e = &policyv1beta1.Eviction{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: ns,
+			},
+		}
+		err = cs.CoreV1().Pods(ns).Evict(e)
+		gomega.Expect(err).Should(gomega.MatchError("Cannot evict pod as it would violate the pod's disruption budget."))
+
+		ginkgo.By("Deleting the pdb to allow a pod to be evicted")
+		deletePDBOrDie(cs, ns, defaultName)
+
+		ginkgo.By("Trying to evict the same pod we tried earlier which should now be evictable")
+		waitForPodsOrDie(cs, ns, 3)
+		err = cs.CoreV1().Pods(ns).Evict(e)
+		framework.ExpectNoError(err) // the eviction is now allowed
+	})
+
+	ginkgo.Context("With another framework", func() {
```

Comment:
Can we change the description here to reflect what we are testing rather than describing how we are doing it? I think what we are checking here is that we can create and list PDBs from different namespaces.